src/javascript/box.js

```diff
@@ -0,0 +1,108 @@
+import API from './api.js';
+
+const api = new API();
+const numberWithCommas = s => String(s).replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+
+export default function Box({ target, buttonTarget, page = 1, count = 5 }) {
+  this.target = target;
+  this.buttonTarget = buttonTarget;
+  this.page = page;
+  this.count = count;
+  this.boxLength = 0;
+  this.boxPrefix = 'box__item';
+  this.buttonPrefix = 'more-button';
+  this.countTarget = null;
+}
+
+Box.prototype = {
+  init,
+  getBoxLength,
+  getBoxList,
+  renderBoxList,
+  getBoxBlock,
+  renderMoreButton,
+  isMoreItem,
+  setDisableMoreButton,
+  updateCount,
+  createMoreEvent,
+}
+
+Box.constructor = Box;
+
+async function init() {
+  this.boxLength = await this.getBoxLength();
+  this.renderMoreButton();
+  await this.renderBoxList();
+  this.createMoreEvent();
+}
+
+async function renderBoxList() {
+  const data = await this.getBoxList();
+  if(data && data.list) {
```

Comment:
Optional chaining 문법을 활용해 볼 수 있겠네요~
https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Optional_chaining

```suggestion
  if (data?.list?.length) {
```